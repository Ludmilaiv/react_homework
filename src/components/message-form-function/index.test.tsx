import { MessageFormFunction } from './';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('MessageFormFunction', () => {
  it('render component', () => {
    render(
      <MessageFormFunction
        addMessage={() => {
          return;
        }}
      />
    );
  });

  it('render with snapshot', () => {
    const { asFragment } = render(
      <MessageFormFunction
        addMessage={() => {
          return;
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('render multiply components', () => {
    render(
      <>
        <MessageFormFunction
          addMessage={() => {
            return;
          }}
        />
        <MessageFormFunction
          addMessage={() => {
            return;
          }}
        />
      </>
    );

    expect(screen.queryAllByRole('form').length).toBe(2);
  });
});
