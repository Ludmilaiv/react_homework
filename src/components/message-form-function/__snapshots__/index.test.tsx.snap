// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageFormFunction render with snapshot 1`] = `
<DocumentFragment>
  <form
    action=""
    class="app__message message-function"
    role="form"
  >
    <input
      class="app__input message-function__input"
      placeholder="Автор"
      type="text"
      value=""
    />
    <input
      class="app__input message-function__input"
      placeholder="Сообщение"
      type="text"
      value=""
    />
    <button
      class="message-function__button"
      disabled=""
      style="border-radius: 3;"
    >
      Отправить
    </button>
  </form>
</DocumentFragment>
`;
